vortex-api/src/main/java/mil/army/usace/hec/vortex/temporal/InstantRecordIndexQuery.java

```diff
@@ -0,0 +1,92 @@
+package mil.army.usace.hec.vortex.temporal;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.logging.Logger;
+
+final class InstantRecordIndexQuery implements RecordIndexQuery {
+    private static final Logger logger = Logger.getLogger(InstantRecordIndexQuery.class.getName());
+    private final NavigableMap<ZonedDateTime, Integer> instantDataTree;
+
+    InstantRecordIndexQuery(List<VortexTimeRecord> recordList) {
```

Comment:
I usually pass an array as an arg to that type is enforced.

Reply:
https://github.com/HydrologicEngineeringCenter/Vortex/pull/116#discussion_r1791894175